subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -411,8 +385,88 @@ uint8_t ll_adv_aux_ad_data_set(uint8_t handle, uint8_t op, uint8_t frag_pref,
 
 #if defined(CONFIG_BT_CTLR_ADV_AUX_PDU_LINK)
 	if ((op == BT_HCI_LE_EXT_ADV_OP_INTERM_FRAG) ||
-	    (op == BT_HCI_LE_EXT_ADV_OP_LAST_FRAG) ||
-	    ad_len_overflow) {
+	    (op == BT_HCI_LE_EXT_ADV_OP_LAST_FRAG)) {
+		/* in the previous step we duplicated the chain
+		 * the next step is to append new data in the last existing pdu in the chain,
+		 */
+
+		val_ptr = hdr_data;
+		*val_ptr++ = len;
+		(void)memcpy(val_ptr, &data, sizeof(data));
+
+		/* Append data to the last PDU */
+		chain_err = ull_adv_aux_pdu_set_clear(adv, pdu_prev, pdu,
+						      ULL_ADV_PDU_HDR_FIELD_AD_DATA_APPEND,
+						      0U, hdr_data);
+
+		LL_ASSERT((!chain_err)  || (chain_err == BT_HCI_ERR_PACKET_TOO_LONG));
+
+		/* FIXME: this probably needs a for loop  */
+		if (chain_err == BT_HCI_ERR_PACKET_TOO_LONG) {
+			/* We could not fit all the data, append as much as possible
+			 * ad_len_overflow_first_try is how much overflows without the AUX ptr
+			 * ad_len_overflow is how much overflows with the AUX ptr
+			 */
+			uint8_t ad_len_overflow_first_try;
+
+			ad_len_overflow_first_try = hdr_data[ULL_ADV_HDR_DATA_DATA_PTR_OFFSET +
+							     ULL_ADV_HDR_DATA_DATA_PTR_SIZE];
+
+			val_ptr = hdr_data;
+			*val_ptr++ = len;
+			(void)memcpy(val_ptr, &data, sizeof(data));
+			val_ptr += sizeof(data);
+			*val_ptr++ = len;
+			(void)memcpy(val_ptr, &data, sizeof(data));
+			chain_err = ull_adv_aux_pdu_set_clear(adv, pdu_prev, pdu,
+							   (ULL_ADV_PDU_HDR_FIELD_AD_DATA_APPEND |
+							    ULL_ADV_PDU_HDR_FIELD_AUX_PTR),
+							    0U, hdr_data);
```

Comment:
One alternative to make this fit in 100 line length is to do
```suggestion
			chain_err = ull_adv_aux_pdu_set_clear(
				adv, pdu_prev, pdu, 
				(ULL_ADV_PDU_HDR_FIELD_AD_DATA_APPEND | 
				 ULL_ADV_PDU_HDR_FIELD_AUX_PTR), 
				0U, hdr_data);1
```

Alternatively move `(ULL_ADV_PDU_HDR_FIELD_AD_DATA_APPEND | ULL_ADV_PDU_HDR_FIELD_AUX_PTR)` to a local `const `variable

Another alternative is to move this into a new function (`ll_adv_aux_ad_data_set` is 500 lines long with this change, so arguably that should be split into multiple functions anyhow).